politeiad/backendv2/tstorebe/plugins/comments/cmds.go

```diff
@@ -0,0 +1,1532 @@
+				score = vote
+			}
+		}
+
+		// Add the net result of all votes from this user to the totals.
+		switch score {
+		case 0:
+			// Nothing to do
+		case -1:
+			downvotes++
+		case 1:
+			upvotes++
+		default:
+			// Something went wrong
+			e := fmt.Errorf("unexpected vote score %v", score)
+			panic(e)
+		}
+	}
+
+	return downvotes, upvotes
+}
+
+func (p *commentsPlugin) cmdNew(treeID int64, token []byte, payload string) (string, error) {
+	log.Tracef("cmdNew: %v %x %v", treeID, token, payload)
+
+	// Decode payload
+	var n comments.New
+	err := json.Unmarshal([]byte(payload), &n)
+	if err != nil {
+		return "", err
+	}
+
+	// Verify token
+	t, err := tokenDecode(n.Token)
+	if err != nil {
+		return "", backend.PluginError{
+			PluginID:     comments.PluginID,
+			ErrorCode:    uint32(comments.ErrorCodeTokenInvalid),
+			ErrorContext: err.Error(),
+		}
+	}
+	if !bytes.Equal(t, token) {
+		e := fmt.Sprintf("comment token does not match route token: "+
+			"got %x, want %x", t, token)
+		return "", backend.PluginError{
+			PluginID:     comments.PluginID,
+			ErrorCode:    uint32(comments.ErrorCodeTokenInvalid),
+			ErrorContext: e,
+		}
+	}
+
+	// Verify signature
+	msg := strconv.FormatUint(uint64(n.State), 10) + n.Token +
+		strconv.FormatUint(uint64(n.ParentID), 10) + n.Comment
+	err = util.VerifySignature(n.Signature, n.PublicKey, msg)
+	if err != nil {
+		return "", convertSignatureError(err)
+	}
+
+	// Verify comment
+	if len(n.Comment) > int(p.commentLengthMax) {
+		e := fmt.Sprintf("max length is %v characters", p.commentLengthMax)
+		return "", backend.PluginError{
+			PluginID:     comments.PluginID,
+			ErrorCode:    uint32(comments.ErrorCodeMaxLengthExceeded),
+			ErrorContext: e,
+		}
+	}
+
+	// Verify record state
+	state, err := p.tstore.RecordState(treeID)
+	if err != nil {
+		return "", err
+	}
+	if uint32(n.State) != uint32(state) {
+		e := fmt.Sprintf("got %v, want %v", n.State, state)
+		return "", backend.PluginError{
+			PluginID:     comments.PluginID,
+			ErrorCode:    uint32(comments.ErrorCodeStateInvalid),
+			ErrorContext: e,
+		}
+	}
+
+	// Get record index
+	ridx, err := p.recordIndex(token, state)
+	if err != nil {
+		return "", err
+	}
+
+	// Verify parent comment exists if set. A parent ID of 0 means that
+	// this is a base level comment, not a reply to another comment.
+	if n.ParentID > 0 && !commentExists(*ridx, n.ParentID) {
+		return "", backend.PluginError{
+			PluginID:     comments.PluginID,
+			ErrorCode:    uint32(comments.ErrorCodeParentIDInvalid),
+			ErrorContext: "parent ID comment not found",
+		}
+	}
+
+	// Setup comment
+	receipt := p.identity.SignMessage([]byte(n.Signature))
+	ca := comments.CommentAdd{
+		UserID:        n.UserID,
+		State:         n.State,
+		Token:         n.Token,
+		ParentID:      n.ParentID,
+		Comment:       n.Comment,
+		PublicKey:     n.PublicKey,
+		Signature:     n.Signature,
+		CommentID:     commentIDLatest(*ridx) + 1,
+		Version:       1,
```

Comment:
Not sure what you mean.  How would a `VersionOne` constant be any different than just hard coding in the 1?  This is the only spot that this will ever appear because there is only one way to create a new comment.

Reply:
Because at some point we need a const when we go to v2 and then we need to find all instances in the code. Using a const helps finding all instances in one go because if you rename it code stops compiling. This is a best practices thing that helps avoiding tears later.

Reply:
You're interpreting the version wrong.  Thats the version of the comment, not of the comment structure.  The comment plugin allows for editing a comment.  When you edit the comment the version gets bumped.  Data versioning is handled in the data descriptors.

https://github.com/lukebp/politeia/blob/tlog/politeiad/backendv2/tstorebe/plugins/comments/cmds.go#L28-L30